uber/templates/regform.html

```diff
@@ -52,8 +52,10 @@
 
     {% if not attendee.is_new %}
         // This removes any badge levels lower than the attendee has purchased already
-        while (_(BADGE_TYPES.options).size() && BADGE_TYPES.options[0].extra < {{ attendee.amount_extra }}) {
-            BADGE_TYPES.options.splice(0, 1);
```

Comment:
Some browsers will still raise an error if you refer to a variable which has not been defined, even when just checking its type to see if it's undefined!  However, the way around this is to go through ``window``.  In case you're not aware, ``x = 5`` is the same as saying ``window.x = 5`` when running Javascript in the browser.  But it's always safe to say ``window.foo`` since that will never raise an exception, and if ``foo`` was never defined it just returns ``undefined``, whereas some browsers will still give you an exception if you say ``foo`` without having defined ``foo`` first.

A better way to write this if statement would therefore be to just say ``if (window.BADGE_TYPES)``, which also checks for ``null``, which is good because we'd get an exception if we tried to enter this if-block when ``BADGE_TYPES`` was null.  (Not that it ever should be, but defensive coding and all that.)